application/src/main/java/org/togetherjava/tjbot/features/help/PinAnswerCommand.java

```diff
@@ -0,0 +1,87 @@
+package org.togetherjava.tjbot.features.help;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.concrete.ForumChannel;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.events.interaction.command.MessageContextInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.config.HelpSystemConfig;
+import org.togetherjava.tjbot.features.BotCommandAdapter;
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.MessageContextCommand;
+
+public final class PinAnswerCommand extends BotCommandAdapter implements MessageContextCommand {
+    private static final String COMMAND_NAME = "pin-answer";
+    private static final int MAX_PINNED_ANSWERS = 10;
+    private final String helpForumPattern;
+
+    public PinAnswerCommand(Config config) {
+        super(Commands.message(COMMAND_NAME), CommandVisibility.GUILD);
+        HelpSystemConfig helpConfig = config.getHelpSystem();
+        helpForumPattern = helpConfig.getHelpForumPattern();
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        Message originalMessage = event.getTarget();
+        User commandInvoker = event.getUser();
+
+        if (!(originalMessage.getChannel() instanceof ThreadChannel threadChannel)) {
+            replyNotInThread(event);
+            return;
+        }
+
+        if (!(threadChannel.getParentChannel() instanceof ForumChannel forumChannel)
+                || !forumChannel.getName().equalsIgnoreCase(helpForumPattern)) {
+            replyNotInQuestionsChannel(event);
+            return;
```

Comment:
like i mentioned in other recent comment, only `replyNotInThread` should be triggered when channel pattern does not match. If you think about it, we only have a single `happy` case that is command is used in `questions`. We don't really care about other cases so we use `replyNotInThread` for the rest of stuff.

Reply:
Oh right, because everything in 'questions' channel is only going to contain threads. There will be no non-threads in the 'questions' channel. That makes sense now.

Reply:
Or I guess I can change the replyNotInThread reply to "This command can only be used in threads in the questions channel" to solve both cases.

Reply:
> Or I guess I can change the replyNotInThread reply to "This command can only be used in threads in the questions channel" to solve both cases.

Yep you're right this gives more context on what went wrong than just "not in thread".